src/python/grpcio_tests/tests/unit/_metadata_code_details_test.py

```diff
@@ -658,6 +658,64 @@ def testCustomCodeReturnNoneStreamUnary(self):
         self.assertEqual(_DETAILS, exception_context.exception.details())
 
 
+class _InspectServicer(_Servicer):
+
+    def __init__(self):
+        super(_InspectServicer, self).__init__()
+        self.actual_code = None
+        self.actual_details = None
+        self.actual_trailing_metadata = None
+
+    def unary_unary(self, request, context):
+        super(_InspectServicer, self).unary_unary(request, context)
+
+        self.actual_code = context.code()
+        self.actual_details = context.details()
+        self.actual_trailing_metadata = context.trailing_metadata()
+
+
+class InspectContextTest(unittest.TestCase):
+
+    def setUp(self):
+        self._servicer = _InspectServicer()
+        self._server = test_common.test_server()
+        self._server.add_generic_rpc_handlers(
+            (_generic_handler(self._servicer),))
+        port = self._server.add_insecure_port('[::]:0')
+        self._server.start()
+
+        self._channel = grpc.insecure_channel('localhost:{}'.format(port))
+        self._unary_unary = self._channel.unary_unary(
+            '/'.join((
+                '',
+                _SERVICE,
+                _UNARY_UNARY,
+            )),
+            request_serializer=_REQUEST_SERIALIZER,
+            response_deserializer=_RESPONSE_DESERIALIZER,
+        )
+
+    def tearDown(self):
+        self._server.stop(None)
+        self._channel.close()
+
+    def testCodeDetailsInContext(self):
+        self._servicer.set_code(_NON_OK_CODE)
+        self._servicer.set_details(_DETAILS)
+
+        with self.assertRaises(grpc.RpcError) as exc_info:
+            self._unary_unary.with_call(object(), metadata=_CLIENT_METADATA)
```

Comment:
nit: We don't need to invoke `with_call` here, since we don't care about the returned value.